src/collectors/file_collector.cpp

```diff
@@ -73,7 +78,8 @@ bayesmix::MarginalState FileCollector::get_state(unsigned int i) {
 }
 
 // \return Chain in deque form
-std::deque<bayesmix::MarginalState> FileCollector::get_chain() {
+template <typename MsgType>
+std::deque<MsgType> FileCollector::get_chain() {
```

Comment:
Remember the thing with the get_chain() method being terrible for the file collector...? Yeah. I think that maybe the time for fixing this by using next_state() has come. Has the time indeed come?

Reply:
I don't exactly get this: do you mean that there is a lot of copy-pasted code between get_chain() and next_state()?

Reply:
No, the thing is that get_chain() should _not_ exist. It only makes sense for the memory collector since the chain is already available. If using the file collector we are building the whole chain from scratch all at once, which makes no sense since we already have it written in the file. You initially proposed a next_state() method to advance of one position when reading the collector, but then it turned out that it was way too complicated for the few time we had and half-assed a solution by creating get_chain(). My question was, should we try to do things properly with next_state() (which is, BTW, currently unused) now that we have time?

Reply:
good point. I'm not sure I remember well why it was problematic.
I think that this is not something super-urgent but we should keep this in mind, hence I'm creating an issue and assigning both of us to that, but will merge this PR without resolving this issue